### PR TITLE
Export logs to log file

### PR DESCRIPTION
**Pull request by MalloZup**
so this PR fix the case where we were not printing to logs to files

### CODE REVIEW

**Comment by MalloZup** [comment]
currently testing it

**Review by stefanotorresi — CHANGES_REQUESTED** [review]
@MalloZup  instead of passing the `$LOG_FILE` variable around into the functions, maybe we could append `| tee -a $log_to_file || log_error "foo"` to each function invocation? This would result in less jumping around the code and all the logging things being in one place.

**Comment by MalloZup** [comment]
@stefanotorresi  can you be more explicit? I didn't understand what you mean tia

File: salt/provision.sh [hunk 1]
```diff
@@ -225,20 +229,20 @@ done
...
 fi
 
+
 if [ $argument_number -eq 0 ]; then
-    bootstrap_salt
-    os_setup
-    predeploy
-    deploy
-    run_tests
+    bootstrap_salt $log_to_file
```
Comment (MalloZup) [inline salt/provision.sh:236]: @stefanotorresi 

`  bootstrap_salt  | tee   $log_to_file || log_error "tests failed"`

Comment (MalloZup) [reply salt/provision.sh:236]: I will check if this work in bash

**Comment by stefanotorresi** [comment]
also, since we are now catching errors from a pipe with `tee`, we need to add the `-o pipefail` option to the script interpreter, so the shebang would become `#!/bin/bash -xeo pipefail` .

File: salt/provision.sh (hunk 1 above)
Comment (stefanotorresi) [reply salt/provision.sh:236]: yes, exactly, so no logging happens inside the `boostrap_salt` function, all of it happens in the "main" of the script.

**Comment by MalloZup** [comment]
I'm ok to close this in favor of @arbulu89  PR. Allthough I find this PR somehow less `bashism` and more understanble , I think for the `provision` script we can live with the other pr if we have comment why we put that command.

**Pull request closed without merging**